gcc/testsuite/rust.test/xfail_compile/continue1.rs

```diff
@@ -5,6 +5,6 @@ fn main() {
     let _fib = {
         continue; // { dg-error "cannot `continue` outside of a loop" }
         // { dg-error "failed to type resolve expression" "" { target { *-*-* } } .-1 }
-        123
+        123 // { dg-warning "unreachable expression" }
```

Comment:
This warning appearing here is wrong-ish. It is true in the sense that it can't be reached, because the program doesn't compile. From a UX perspective this produces unnecessary clutter though. This shouldn't block the PR from going through.

Reply:
Yeah, you are right. This code cannot compile with rustc.

Reply:
But I won't deal with this (at least in this PR). I've explained the reason below, see https://github.com/Rust-GCC/gccrs/pull/364#discussion_r613216808. (Note removing unreachable statements/expressions happens eariler than type check.)